### PR TITLE
Feature/driver registration flow

### PR DESCRIPTION
**Pull request by akhtyamovrr**

### CODE REVIEW

File: drivers/firebase.go [hunk 1]
```diff
@@ -43,7 +47,21 @@ type Firebase struct {
 	collection      *Collection
 }
 
-func NewFirebase(ctx context.Context, config *FirebaseConfig, collection *Collection) (*Firebase, error) {
+func init() {
+	if err := RegisterDriverConstructor(firebaseType, NewFirebase); err != nil {
+		logging.Errorf("Failed to register driver %s: %s", firebaseType, err)
```
Comment (xtreding) [inline drivers/firebase.go:52]: %s -> %v in error case

File: drivers/google_analytics.go [hunk 2]
```diff
@@ -70,9 +66,23 @@ type GoogleAnalytics struct {
 	reportFieldsConfig *ReportFieldsConfig
 }
 
-func NewGoogleAnalytics(ctx context.Context, config *GoogleAnalyticsConfig, collection *Collection) (*GoogleAnalytics, error) {
+func init() {
+	if err := RegisterDriverConstructor(googleAnalyticsType, NewGoogleAnalytics); err != nil {
+		logging.Errorf("Failed to register driver %s: %s", googleAnalyticsType, err)
```
Comment (xtreding) [inline drivers/google_analytics.go:71]: same

File: drivers/google_play.go [hunk 3]
```diff
@@ -67,7 +69,21 @@ type GooglePlay struct {
 	collection *Collection
 }
 
-func NewGooglePlay(ctx context.Context, config *GooglePlayConfig, collection *Collection) (*GooglePlay, error) {
+func init() {
+	if err := RegisterDriverConstructor(googlePlayType, NewGooglePlay); err != nil {
+		logging.Errorf("Failed to register driver %s: %s", googlePlayType, err)
```
Comment (xtreding) [inline drivers/google_play.go:74]: same

**Review by xtreding — APPROVED** [review]

File: drivers/firebase.go (hunk 1 above)
Comment (akhtyamovrr) [reply drivers/firebase.go]: fixed

File: drivers/google_play.go (hunk 3 above)
Comment (akhtyamovrr) [reply drivers/google_play.go]: fixed

File: drivers/google_analytics.go (hunk 2 above)
Comment (akhtyamovrr) [reply drivers/google_analytics.go]: fixed

**Pull request merged**